ledger/accountdb.go

```diff
@@ -1517,12 +1517,17 @@ func (ba *baseAccountData) GetAccountData() basics.AccountData {
 }
 
 // IsEmpty returns true if all of the fields other than UpdateRound are zero.
-func (bv baseVotingData) IsEmpty() bool {
-	return bv == baseVotingData{}
+func (bv *baseVotingData) IsEmpty() bool {
+	return bv.VoteID == crypto.OneTimeSignatureVerifier{} &&
+		bv.SelectionID == crypto.VRFVerifier{} &&
+		bv.VoteFirstValid == basics.Round(0) &&
+		bv.VoteLastValid == basics.Round(0) &&
+		bv.VoteKeyDilution == 0 &&
+		bv.StateProofID == merklesignature.Verifier{}
```

Comment:
I'm confused about what happened here. I recall the discussion, but don't recall how the plain equality check (that you've reverted here) ended up being used. It seems wrong, given the comment, so I'm in favor of this change.  But I wonder if callers were changed somehow to make the simple check correct?

I also wonder if Will's suggestion, which I liked a lot for staying correct in the face of baseVotingData changes, is fast enough to consider - maybe things changed with the pointer/value swap.  He had suggested we implement this as an `==` check against: `baseVotingData{UpdateRound: bv.UpdateRound}` which nicely shows we want to check everything except the UpdateRound, and needs no updates.

Reply:
oh, ok, then yes, we can just use the simple `==` here.

Reply:
restored to old version